frontend/src/components/ui/field.tsx

```diff
@@ -0,0 +1,18 @@
+import {ComponentPropsWithoutRef, Ref, forwardRef} from "react"
+import {cn} from "~/lib/utils";
+import {useTheme} from "next-themes";
+
+type HtmlInputProps = ComponentPropsWithoutRef<'input'>
+
+interface InputProps extends HtmlInputProps {
+  label: string;
+  error?: string;
+}
+
+export const Field = forwardRef<HTMLInputElement, InputProps>( function Field({error, label, ...props}: InputProps, ref) {
```

Comment:
Btw Shadcn has some 🔥 components like Input and Form that take care of that boilerplate for you

Reply:
I'll look more into shadcn